backend/portal/api/curation/v1/curation/collections/collection_id/datasets/dataset_id/actions.py

```diff
@@ -48,7 +48,17 @@ def get(collection_id: str, dataset_id: str = None):
     if version is None:
         raise NotFoundHTTPException("Dataset not found")
 
-    response_body = reshape_dataset_for_curation_api(version, collection_version.published_at is not None)
+    # A canonical url should be only used in two cases:
+    # 1. the collection version is unpublished but it's not a revision
+    # 2. the collection version is published
+    should_use_canonical_url = (
```

Comment:
should seems wishy washy and extra. Change to:
```suggestion
    use_canonical_url = (
```